ci/scripts/deploy-cluster.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+set -eoux pipefail
+
+log() {
+  local message="${1:-""}"
+  echo -e "\\033[1;33m${message}\\033[0m"
+}
+
+# It returns the location where most instances are available for the instance
+# type passed as parameter.
+#
+# It returns 0 if it found one or 1 if it didn't. On return it sets the
+# $AVAIL_LOCATIONS variable.
+best_location_for_instance_type() {
+    local instance_type="${1}"
+
+    set +x
+    AVAIL_LOCATIONS=$(curl -X GET --header 'Accept: application/json' --header "X-Auth-Token: $PACKET_AUTH_TOKEN" 'https://api.packet.net/capacity' |
+        jq -r '.capacity | to_entries[] | { region: .key, avail: .value."'"${instance_type}"'".level} | select(.avail == "normal") | { region: .region } | values[]')
+    set -x
+
+    if [ -n "$AVAIL_LOCATIONS" ]; then
+        return 0
+    fi
+
+    echo "WARNING: no region with \"normal\" ${instance_type} capacity found. Trying \"limited\" capacity..."
+
+    set +x
+    AVAIL_LOCATIONS=$(curl -X GET --header 'Accept: application/json' --header "X-Auth-Token: $PACKET_AUTH_TOKEN" 'https://api.packet.net/capacity' |
+        jq -r '.capacity | to_entries[] | { region: .key, avail: .value."'"${instance_type}"'".level} | select(.avail == "limited") | { region: .region } | values[]')
+    set -x
+
+    if [ -n "$AVAIL_LOCATIONS" ]; then
+        return 0
+    fi
+
+    echo "ERROR: no region with ${instance_type} availability"
+    return 1
+}
+
+# =======================================================================
+
+if [ "${platform}" != baremetal ] && [ "${platform}" != "kvm-libvirt" ]; then
```

Comment:
Would be nice to have some safety switch to avoid running this script locally. I think it may override user's SSH private key...

Reply:
We can check if `~/.ssh/id_rsa` exists already?

Reply:
I'd use some env variable like `CI` on top, to only be able to run when it's set. And probably Makefile itself shouldn't set it.

Reply:
I added the variable `CI`. There is no point in checking its value because it will fail when we try to access because `-u` is set.